pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/NullHandlingIntegrationTest.java

```diff
@@ -168,4 +168,18 @@ public void testCaseWithNotNullDescription()
     String query = "SELECT CASE WHEN description IS NOT NULL THEN 1 ELSE 0 END FROM " + getTableName();
     testQuery(query);
   }
+
+  @Test
+  public void testCaseWithIsDistinctFrom()
+      throws Exception {
+    String query = "SELECT salary IS DISTINCT FROM salary FROM " + getTableName();
```

Comment:
Can we get a couple more test cases using "IS [NOT] DISTINCT FROM" operator over expressions and in where clause?

Reply:
Added a test in where clause. This doesn't work over expressions because we only support null evaluation on column names.